### PR TITLE
Add the ability for Script based commands to specify their "repeat command"

### PR DESCRIPTION
**Pull request by jimingham**
Among other things, returning an empty string as the repeat command disables auto-repeat, which can be useful for state-changing commands.

There's one remaining refinement to this setup, which is that for parsed script commands, it should be possible to change an option value, or add a new option value that wasn't originally specified, then ask lldb "make this back into a command string".  That would make doing fancy things with repeat commands easier.

That capability isn't present in the lldb_private side either, however.  So that's for a next iteration.

I haven't added this to the docs on adding commands yet.  I wanted to make sure this was an acceptable approach before I spend the time to do that.

### CODE REVIEW

File: lldb/bindings/python/python-wrapper.swig [hunk 1]
```diff
@@ -831,6 +831,29 @@ bool lldb_private::python::SWIGBridge::LLDBSwigPythonCallCommandObject(
...
+
+  PythonObject self(PyRefType::Borrowed, implementor);
+  auto pfunc = self.ResolveName<PythonCallable>("get_repeat_command");
+  // If not implemented, repeat the exact command.
+  if (!pfunc.IsAllocated())
+    return std::nullopt;
+
+  PythonObject result;
+  PythonString command_str(command);
+  result = pfunc(command_str);
```
Comment (medismailben) [inline lldb/bindings/python/python-wrapper.swig:847]: nit:
```suggestion
  PythonString command_str(command);
  PythonObject result = pfunc(command_str);
```

File: lldb/source/Commands/CommandObjectCommands.cpp [hunk 2]
```diff
@@ -1827,6 +1838,15 @@ class CommandObjectScriptingObjectParsed : public CommandObjectParsed {
...
 
+  std::optional<std::string> GetRepeatCommand(Args &args,
+                                              uint32_t index) override {
+    ScriptInterpreter *scripter = GetDebugger().GetScriptInterpreter();
+    if (!scripter)
+      return std::nullopt;
+
+    return scripter->GetRepeatCommandForScriptedCommand(m_cmd_obj_sp, args);
+  }
+
```
Comment (medismailben) [inline lldb/source/Commands/CommandObjectCommands.cpp:1849]: nit: This looks pretty much the same as the one in `CommandObjectScriptingObjectRaw`. I understand that these have different base classes but I guess we could use multiple inheritance (in a follow-up) to refactor the 2 classes and reduce code duplication.

File: lldb/test/API/commands/command/script/add/TestAddParsedCommand.py [hunk 3]
```diff
@@ -32,6 +37,39 @@ def check_help_options(self, cmd_name, opt_list, substrs=[]):
...
+        )
+
+        in_fileH.close()
+        out_fileH.close()
+
+        results = None
+        with open(self.stdout_path, "r") as out_fileH:
+            results = out_fileH.read()
+
+        print(f"RESULTS:\n{results}\nDONE")
```
Comment (medismailben) [inline lldb/test/API/commands/command/script/add/TestAddParsedCommand.py:68]: May be we should only print if the test suite in verbose/trace mode.

**Review by medismailben — APPROVED** [review]
LGTM with nits

File: lldb/source/Commands/CommandObjectCommands.cpp (hunk 2 above)
Comment (jimingham) [reply lldb/source/Commands/CommandObjectCommands.cpp:1849]: I was holding off on refactoring these classes till we're switching to the Scripted Interface infrastructure.

File: lldb/test/API/commands/command/script/add/TestAddParsedCommand.py (hunk 3 above)
Comment (jimingham) [reply lldb/test/API/commands/command/script/add/TestAddParsedCommand.py:68]: Doh.  Probably can just get rid of this.

**Review by medismailben — APPROVED** [review]
LGTM!

**Pull request merged**